tests/integration/long_callback/app_progress_delete.py

```diff
@@ -36,7 +36,9 @@
 )
 def on_bg_progress(set_progress, _):
     set_progress("start")
-    time.sleep(2)
+    time.sleep(1)
```

Comment:
Might be better to change this to lock to make sure it's not reset.

Reply:
I'm assuming you meant to write _look_ instead of _lock_. Would 7437e2b work?

The final results for this test (_tests/integration/long_callback/test_basic_long_callback014.py_) would be:
- Using `dev`
![image](https://github.com/plotly/dash/assets/33813815/8a88dcc4-878c-4bf2-ad81-e8ee0c729165)
- Using this PR
![image](https://github.com/plotly/dash/assets/33813815/b59852a7-5a67-4d10-8d7f-49bfd7b3d785)

In `dev`, the `#progress-counter` is _3_ because the value gets incremented every time `set_progress` is called, but it's correctly being left as _2_ with this PR's changes.

As for why `#progress-output` is set back to _start_ in `dev`, I am not sure 🤔. However, it's correctly being left as _stop_ in this PR.

Reply:
I think @T4rk1n was referring to `multiprocessing.Lock`, which you can find in some of the other Dash tests. This can remove the need for any explicit `sleep` statements, which not only makes the tests faster but also more robust on CI where you never know what else is running on the same hardware and can lead to unexpected delays that will intermittently break tests based on `sleep`.

Reply:
Yes, I was talking about a `multiprocessing.Lock`.

Basically you would do in the callback

```
with lock:
    set_progress(no_update)
```

Then in the test you can also request the lock and do the assertions, so you can be sure the call to set_progress is done before the assertions.

Reply:
Ohh I see, that makes sense. I added the lock in 21bc4b2 🔒